origins_workflow/src/Controller/ModerationStateController.php

```diff
@@ -69,6 +92,15 @@ public function changeState($nid, $new_state) {
           '@user' => $this->currentUser()->getAccountName(),
         ]);
         $this->logger->notice($message);
+
+        if (!empty(\Drupal::request()->query->get('confirm'))) {
+          $message = t('Moderation state of "@title" changed to @new_state', [
+            '@title' => $entity->getTitle(),
+            '@new_state' => $new_state_entity->label(),
+          ]);
+
+          \Drupal::messenger()->addMessage($message, \Drupal::messenger()::TYPE_STATUS);
```

Comment:
Ah, this needs to be passed in by dependency injection. I'll sort that before merging!

Reply:
Added services for messenger and also request stack